src/Expander/Primitives.hs

```diff
@@ -675,27 +623,8 @@ elsePattern (Right dest) stx = do
 addDatatype :: Ident -> Datatype -> [Kind] -> Expand ()
 addDatatype name dt argKinds = do
   name' <- addRootScope' name
-  let implType =
-        \k dest stx -> do
-          Stx _ _ (me, args) <- mustBeCons stx
-          _ <- mustBeIdent me
```

Comment:
We are no longer checking that the list is identifier-headed. Seems redundant to me; how would we get here if not because we are expanding an identifier-headed list whose head identifier names a datatype?

Reply:
I think that this was overzealous parsing and was never needed